Survivl_model object, plasmode, and vignettes

chasehmathis opened the pull request:
### Survivl_model 
- Created a mirror image of the causl_model object.
- not experienced with the S3 methods so print.survivl_model and modify.survivl_model aren't perfectly working

### Plasmode
- previously we had to provide quantiles to do plasmode for survivl modeling. 
- Now, we use the empirical rank technique from causl package and the linear regression to residualize away correlations
- Small issues make it challenging to plasmode on top of a any arbitrary t point, working with just 0 now

### Vignettes
- Vignettes haven't been updated since 2022
- Got rid of the censoring because I believe our method now with inversion technique consider censoring to be a competing risk which is more complicated .Either way, the code now doesn't not work with the censoring example.
- Switched all msm_samp to survivl_model and then calling rmsm(n, surv_model)
- switched some ipw calculations from by hand to using ipw::ipwtm. Makes it a bit easier to read. Also using coxph instead of svyglm

chasehmathis committed:
new survivl object. tests passing

chasehmathis committed:
causl quantile logic to plasmode in survivl

chasehmathis committed:
Added collect_events for future usage

chasehmathis committed:
Removing censoring vignette for now. need to adapt new logic for use

chasehmathis committed:
Fixing all vignettes to correspond to new class object

chasehmathis committed:
updating website

rje42 reviewed (changes requested):
Can you reuse `rmsm()` in `msm_samp()`?  And also I recommend pushing this to the `devel` branch, rather than the main one.

chasehmathis committed:
New tests, vignettes, and small fixes

chasehmathis committed:
Added rmsm in msm_samp

chasehmathis committed:
Building site for documentation

chasehmathis commented:
- Added rmsm to msm_samp and extended support for non-survival longitudinal outcomes (Section 4 in the recent paper), which was missing in the original code. 
- I didn’t see this functionality in causl, so I implemented it here. 
- While it’s not a survival outcome, I added it to the survivl package since it already handles longitudinal data. 
- Moving it to causl would require building out additional infrastructure, I think?

- Also, Attempting to create a pull request into the devel branch crashes GitHub on my end.

chasehmathis committed:
package passing with new causl copula functions

chasehmathis committed:
survivl still good with new causl updates

Commit:
allowing for more complex copulas in the longitudinal sense

chasehmathis committed:
complicated copulas

chasehmathis committed:
Simulation via seaman + keogh

chasehmathis committed:
vignettes + site for continuous seaman+keogh. Issues with kable

rje42 committed:
Rbuildignore

rje42 committed:
Merge branch 'main' of github.com:chasehmathis/survivl

chasehmathis committed:
post-robin meeting comit

chasehmathis committed:
Merge branch 'main' of github.com:chasehmathis/survivl

rje42 committed:
Improved formating
